patroni/postgresql.py

```diff
@@ -1400,7 +1403,16 @@ def rewind(self, leader):
         # make it store the new timeline (5540277D.8020309@iki.fi)
         leader_status = self.checkpoint(r)
         if leader_status:
-            return logger.warning('Can not use %s for rewind: %s', leader.name, leader_status)
+            logger.warning('Can not use %s for rewind: %s', leader.name, leader_status)
+            for name in ('remove_data_directory_on_rewind_failure', 'remove_data_directory_on_diverged_timelines'):
```

Comment:
This loop is absolutely the same as a few lines below: https://github.com/zalando/patroni/pull/972/files#diff-d8c1582337c8426ef750420dc6b76949R1424

I think it makes sense to move it into the separate method to avoid code duplication.